packages/apps/dashboard/ui-2024/src/components/SearchBar/SearchBar.tsx

```diff
@@ -30,58 +30,50 @@ import {
 
 interface SearchBarProps {
   className?: string;
-  displaySearchBar?: boolean;
-  isTopBar?: boolean;
   initialInputValue?: string;
 }
 
 const SearchBar: FC<SearchBarProps> = ({
   className = '',
-  displaySearchBar,
   initialInputValue = '',
-  isTopBar = false,
 }) => {
   const { mobile } = useBreakPoints();
   const [inputValue, setInputValue] = useState<string>(initialInputValue);
-  const [selectValue, setSelectValue] = useState<number | null>(null);
+  const [selectValue, setSelectValue] = useState<number | string>('');
   const [focus, setFocus] = useState<boolean>(false);
-  const { filterParams, setAddress, setChainId } = useWalletSearch();
+  const { filterParams } = useWalletSearch();
   const navigate = useNavigate();
 
   const navigateToAddress = useCallback(
     (chainIdParam?: number | undefined) => {
-      const chainId = chainIdParam || filterParams.chainId || -1;
-      const address = (isTopBar ? filterParams.address : inputValue) || '';
+      const chainId = chainIdParam || selectValue || -1;
+      const address = inputValue || '';
       navigate(`/search/${chainId}/${address}`);
     },
-    [filterParams.address, filterParams.chainId, inputValue, isTopBar, navigate]
+    [inputValue, selectValue, navigate]
   );
 
   useEffect(() => {
     const networkName = getNetwork(filterParams.chainId || -1)?.name || '';
     if (networkName) {
       setSelectValue(filterParams.chainId);
     }
-  }, [filterParams.chainId, navigateToAddress]);
+  }, [filterParams.chainId]);
 
   useEffect(() => {
     setInputValue(filterParams.address);
   }, [filterParams.address]);
 
   const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (isTopBar) {
-      setAddress(event.target.value);
-    } else {
-      setInputValue(event.target.value);
-    }
+    setInputValue(event.target.value);
   };
 
-  const handleSelectChange = (event: SelectChangeEvent<number | null>) => {
+  const handleSelectChange = (event: SelectChangeEvent<number | string>) => {
     const chainId = Number(event.target.value);
-    if (inputValue && !!inputValue.length) {
+    const isAnyAddress = inputValue && !!inputValue.trim().length;
+    setSelectValue(chainId);
+    if (isAnyAddress) {
       navigateToAddress(chainId);
```

Comment:
Let's  remove this and trigger "search" only when when form submitted (i.e. search icon click or "enter" key after providing input

Reply:
@dnechay Yes its better now :) done